src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/PayPalFlowUtils.res

```diff
@@ -0,0 +1,167 @@
+      subText: "PayPal denied your application to use Advanced Credit and Debit Card Payments.",
+      refreshStatusText: "Try again?",
+    }
+  | More_permissions_needed => {
+      headerText: "PayPal requires you to grant all permissions",
+      subText: "You need to grant all the permissions to create and receive payments. Please click on the Signup to PayPal button and grant the permissions.",
+      buttonText: "Complete Signing up",
+      refreshStatusText: "Already granted all permissions?",
+    }
+
+  | Email_not_verified => {
+      headerText: "Your email is yet to be confirmed!",
+      subText: "Please confirm your email address on https://www.paypal.com/businessprofile/settings in order to receive payments.",
+      refreshStatusText: "Email already confirmed?",
+    }
+  | _ => {
+      headerText: "",
+      subText: "",
+    }
+  }
+}
+
+let stringToVariantMapper = strValue => {
+  open PayPalFlowTypes
+  switch strValue {
+  | "account_not_found" => Account_not_found
+  | "payments_not_receivables" => Payments_not_receivable
+  | "ppcp_custom_denied" => Ppcp_custom_denied
+  | "more_permissions_needed" => More_permissions_needed
+  | "email_not_verified" => Email_not_verified
+  | "connector_integrated" => Connector_integrated
+  | _ => Account_not_found
+  }
+}
+
+let paypalAPICall = async (~updateDetails, ~connectorId, ~profileId) => {
+  open APIUtils
+  try {
+    let paypalBody =
+      [
+        ("connector", "paypal"->Js.Json.string),
+        ("connector_id", connectorId->Js.Json.string),
+        ("profile_id", profileId->Js.Json.string),
+      ]->LogicUtils.getJsonFromArrayOfJson
+
+    let url = `${getURL(~entityName=PAYPAL_ONBOARDING, ~methodType=Post, ())}/sync`
+    let response = await updateDetails(url, paypalBody, Fetch.Post)
+
+    response->LogicUtils.getDictFromJsonObject->LogicUtils.getJsonObjectFromDict("paypal")
+  } catch {
+  | _ => Js.Json.null
+  }
+}
+
+let paypalAccountStatusAtom: Recoil.recoilAtom<PayPalFlowTypes.setupAccountStatus> = Recoil.atom(.
+  "paypalAccountStatusAtom",
+  PayPalFlowTypes.Account_not_found,
+)
+
+let handleConnectorIntegrated = (
+  ~dictValue,
+  ~setInitialValues,
+  ~connector,
+  ~handleStateToNextPage,
+) => {
+  let values = dictValue->LogicUtils.getJsonObjectFromDict("connector_integrated")
+  let bodyTypeValue =
+    dictValue
+    ->LogicUtils.getDictfromDict("connector_integrated")
+    ->LogicUtils.getDictfromDict("connector_account_details")
+    ->LogicUtils.getString("auth_type", "")
+  let body = ConnectorUtils.generateInitialValuesDict(
+    ~values,
+    ~connector,
+    ~bodyType=bodyTypeValue,
+    ~isPayoutFlow=false,
+    (),
+  )
+  setInitialValues(_ => body)
+  handleStateToNextPage()->ignore
+}
+
+let handleObjectResponse = (
+  ~dict,
+  ~setSetupAccountStatus,
+  ~setInitialValues,
+  ~connector,
+  ~handleStateToNextPage,
+) => {
+  let dictkey = dict->Js.Dict.keys->Belt.Array.get(0)->Belt.Option.getWithDefault("")
+
```

Comment:
why are you checking for the zeroth `Belt.Array.get(0)` index?

Reply:
The response consists of a key-value pair, and depending on the key, specific actions need to be executed. There will always be one key in the response.